0.18 build fails

### Description

I am trying to build a previously working project with buffalo 0.18 and it fails (0.17.x was fine)

### Steps to Reproduce the Problem

```
buffalo build
```

### Expected Behavior

no errors

### Actual Behavior

```
golang_test               | DEBU[2021-11-26T08:56:19Z] Exec: go build -v -tags development -o bin/test -ldflags -linkmode external -extldflags "-static"
golang_test               | test/a
golang_test               | test/migrations
golang_test               | test
golang_test               | # test
golang_test               | ./buffalo_build_main.go:13:2: imported and not used: "test/models"
golang_test               | ./buffalo_build_main.go:14:2: imported and not used: "test/migrations"
golang_test               | DEBU[2021-11-26T08:56:20Z] Rollback: main.go
golang_test               | DEBU[2021-11-26T08:56:20Z] File: /test/test/main.go
golang_test               | DEBU[2021-11-26T08:56:20Z] Delete: a/aa.go
golang_test               | DEBU[2021-11-26T08:56:20Z] Delete: a/embed.go
golang_test               | DEBU[2021-11-26T08:56:20Z] Delete: a/inflections.json
golang_test               | DEBU[2021-11-26T08:56:20Z] Delete: buffalo_build_main.go
golang_test               | DEBU[2021-11-26T08:56:20Z] Delete: migrations/.pop-tmp.md
golang_test               | DEBU[2021-11-26T08:56:20Z] Delete: migrations/embed.go
golang_test               | DEBU[2021-11-26T08:56:20Z] Exec: go mod tidy
golang_test               | Usage:
golang_test               |   buffalo build [flags]
golang_test               |
golang_test               | Aliases:
golang_test               |   build, b, bill, install
golang_test               |
golang_test               | Flags:
golang_test               |       --clean-assets               will delete public/assets before calling webpack
golang_test               |       --dry-run                    runs the build 'dry'
golang_test               |       --environment string         set the environment for the binary (default "development")
golang_test               |   -e, --extract-assets             extract the assets and put them in a distinct archive
golang_test               |   -h, --help                       help for build
golang_test               |       --ldflags string             set any ldflags to be passed to the go build
golang_test               |       --mod string                 -mod flag for go build
golang_test               |   -o, --output string              set the name of the binary
golang_test               |   -k, --skip-assets                skip running webpack and building assets
golang_test               |       --skip-build-deps            skip building dependencies
golang_test               |       --skip-template-validation   skip validating templates
golang_test               |   -s, --static                     build a static binary using  --ldflags '-linkmode external -extldflags "-static"'
golang_test               |   -t, --tags string                compile with specific build tags
golang_test               |   -v, --verbose                    print debugging information
golang_test               |
golang_test               | ERRO[0001] Error: exit status 2
```
### Info

Please run `buffalo info` and paste the information below where it says "PASTE_HERE".

<details>

```
-> Go: Checking installation
✓ The `go` executable was found on your system at: /usr/local/go/bin/go

-> Go: Checking minimum version requirements
✓ Your version of Go, 1.17.3, meets the minimum requirements.

-> Go: Checking Package Management
✓ You are using Go Modules (`go`) for package management.

-> Go: Checking PATH
✓ Your PATH contains /go/bin.

-> Node: Checking installation
✘ The `node` executable could not be found on your system.
For help setting up your Node environment please follow the instructions for you platform at:

https://nodejs.org/en/download/

-> NPM: Checking installation
✘ The `npm` executable could not be found on your system.
For help setting up your NPM environment please follow the instructions for you platform at:

https://docs.npmjs.com/getting-started/configuring-your-local-environment

-> Yarn: Checking installation
✘ The `yarnpkg` executable could not be found on your system.
For help setting up your Yarn environment please follow the instructions for you platform at:

https://yarnpkg.com/en/docs/install

-> PostgreSQL: Checking installation
✘ The `postgres` executable could not be found on your system.
For help setting up your Postgres environment please follow the instructions for you platform at:

https://www.postgresql.org/download/

-> MySQL: Checking installation
✘ The `mysql` executable could not be found on your system.
For help setting up your MySQL environment please follow the instructions for you platform at:

https://www.mysql.com/downloads/

-> SQLite3: Checking installation
✘ The `sqlite3` executable could not be found on your system.
For help setting up your SQLite3 environment please follow the instructions for you platform at:

https://www.sqlite.org/download.html

-> Cockroach: Checking installation
✘ The `cockroach` executable could not be found on your system.
For help setting up your Cockroach environment please follow the instructions for you platform at:

https://www.cockroachlabs.com/docs/stable/

-> Buffalo (CLI): Checking installation
✓ The `buffalo` executable was found on your system at: /go/bin/buffalo

-> Buffalo (CLI): Checking minimum version requirements
✓ Your version of Buffalo (CLI), v0.18.0, meets the minimum requirements.

-> Buffalo: Application Details
Pwd         /test/test
Root        /test/test
GoPath      /go
PackagePkg  test
ActionsPkg  test/actions
ModelsPkg   test/models
GriftsPkg   test/grifts
WithModules true
Name        test
Bin         bin/test
VCS         git
WithPop     false
WithSQLite  false
WithDep     false
WithWebpack false
WithNodeJs  false
WithYarn    false
WithDocker  true
WithGrifts  false
AsWeb       false
AsAPI       true
InApp       true
PackageJSON {map[]}

-> Buffalo: config/buffalo-app.toml
name = "test"
bin = "bin/test"
vcs = "git"
with_pop = false
with_sqlite = false
with_dep = false
with_webpack = false
with_nodejs = false
with_yarn = false
with_docker = true
with_grifts = false
as_web = false
as_api = true

-> Buffalo: config/config.go
package config
[...]

-> Buffalo: go.mod
module test

go 1.17

require (
	github.com/go-sql-driver/mysql v1.6.0
	github.com/gobuffalo/buffalo v0.18.0
	github.com/gobuffalo/envy v1.10.1
	github.com/gobuffalo/httptest v1.5.1
	github.com/gobuffalo/mw-contenttype v0.0.0-20190224202710-36c73cc938f3
	github.com/gobuffalo/mw-paramlogger v0.0.0-20190224201358-0d45762ab655
	github.com/gobuffalo/nulls v0.4.1
	github.com/gobuffalo/suite v2.8.2+incompatible
	github.com/gobuffalo/validate/v3 v3.3.1 // indirect
	github.com/gobuffalo/x v0.0.0-20190224155809-6bb134105960
	github.com/google/uuid v1.1.2
	github.com/jmoiron/sqlx v1.3.4
	github.com/pkg/errors v0.9.1
	github.com/rs/cors v1.6.0
	go.uber.org/zap v1.17.0
	golang.org/x/sys v0.0.0-20211124211545-fe61309f8881 // indirect
	gopkg.in/h2non/gock.v1 v1.0.15
)

require (
	github.com/BurntSushi/toml v0.4.1 // indirect
	github.com/Masterminds/semver/v3 v3.1.1 // indirect
	github.com/auth0/go-jwt-middleware v1.0.1 // indirect
	github.com/aymerick/douceur v0.2.0 // indirect
	github.com/cockroachdb/cockroach-go v0.0.0-20190925194419-606b3d062051 // indirect
	github.com/davecgh/go-spew v1.1.1 // indirect
	github.com/dustin/go-humanize v1.0.0 // indirect
	github.com/fatih/color v1.13.0 // indirect
	github.com/fatih/structs v1.1.0 // indirect
	github.com/felixge/httpsnoop v1.0.2 // indirect
	github.com/form3tech-oss/jwt-go v3.2.2+incompatible // indirect
	github.com/fsnotify/fsnotify v1.5.1 // indirect
	github.com/go-testfixtures/testfixtures/v3 v3.4.0 // indirect
	github.com/gobuffalo/events v1.4.2 // indirect
	github.com/gobuffalo/fizz v1.14.0 // indirect
	github.com/gobuffalo/flect v0.2.4 // indirect
	github.com/gobuffalo/genny v0.6.0 // indirect
	github.com/gobuffalo/github_flavored_markdown v1.1.1 // indirect
	github.com/gobuffalo/helpers v0.6.4 // indirect
	github.com/gobuffalo/logger v1.0.6 // indirect
	github.com/gobuffalo/meta v0.3.1 // indirect
	github.com/gobuffalo/mw-csrf v0.0.0-20190129204204-25460a055517 // indirect
	github.com/gobuffalo/packd v1.0.1 // indirect
	github.com/gobuffalo/plush v3.8.3+incompatible // indirect
	github.com/gobuffalo/plush/v4 v4.1.9 // indirect
	github.com/gobuffalo/pop v4.13.1+incompatible // indirect
	github.com/gobuffalo/pop/v6 v6.0.0 // indirect
	github.com/gobuffalo/tags/v3 v3.1.2 // indirect
	github.com/gobuffalo/validate v2.0.4+incompatible // indirect
	github.com/gofrs/uuid v4.2.0+incompatible // indirect
	github.com/gorilla/css v1.0.0 // indirect
	github.com/gorilla/handlers v1.5.1 // indirect
	github.com/gorilla/mux v1.8.0 // indirect
	github.com/gorilla/securecookie v1.1.1 // indirect
	github.com/gorilla/sessions v1.2.1 // indirect
	github.com/h2non/parth v0.0.0-20190131123155-b4df798d6542 // indirect
	github.com/inconshreveable/mousetrap v1.0.0 // indirect
	github.com/jackc/chunkreader/v2 v2.0.1 // indirect
	github.com/jackc/pgconn v1.10.1 // indirect
	github.com/jackc/pgio v1.0.0 // indirect
	github.com/jackc/pgpassfile v1.0.0 // indirect
	github.com/jackc/pgproto3/v2 v2.2.0 // indirect
	github.com/jackc/pgservicefile v0.0.0-20200714003250-2b9c44734f2b // indirect
	github.com/jackc/pgtype v1.9.0 // indirect
	github.com/jackc/pgx/v4 v4.14.0 // indirect
	github.com/joho/godotenv v1.4.0 // indirect
	github.com/karrick/godirwalk v1.16.1 // indirect
	github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51 // indirect
	github.com/lib/pq v1.10.4 // indirect
	github.com/luna-duclos/instrumentedsql v1.1.3 // indirect
	github.com/markbates/grift v1.5.0 // indirect
	github.com/markbates/oncer v1.0.0 // indirect
	github.com/markbates/refresh v1.12.0 // indirect
	github.com/markbates/safe v1.0.1 // indirect
	github.com/markbates/sigtx v1.0.0 // indirect
	github.com/mattn/go-colorable v0.1.12 // indirect
	github.com/mattn/go-isatty v0.0.14 // indirect
	github.com/mattn/go-sqlite3 v2.0.3+incompatible // indirect
	github.com/microcosm-cc/bluemonday v1.0.16 // indirect
	github.com/mitchellh/go-homedir v1.1.0 // indirect
	github.com/monoculum/formam v3.5.5+incompatible // indirect
	github.com/pmezard/go-difflib v1.0.0 // indirect
	github.com/rogpeppe/go-internal v1.8.0 // indirect
	github.com/sergi/go-diff v1.2.0 // indirect
	github.com/sirupsen/logrus v1.8.1 // indirect
	github.com/sourcegraph/annotate v0.0.0-20160123013949-f4cad6c6324d // indirect
	github.com/sourcegraph/syntaxhighlight v0.0.0-20170531221838-bd320f5d308e // indirect
	github.com/spf13/cobra v1.2.1 // indirect
	github.com/spf13/pflag v1.0.5 // indirect
	github.com/stretchr/testify v1.7.0 // indirect
	go.uber.org/atomic v1.7.0 // indirect
	go.uber.org/multierr v1.6.0 // indirect
	golang.org/x/crypto v0.0.0-20211117183948-ae814b36b871 // indirect
	golang.org/x/net v0.0.0-20211123203042-d83791d6bcd9 // indirect
	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c // indirect
	golang.org/x/term v0.0.0-20210927222741-03fcf44c2211 // indirect
	golang.org/x/text v0.3.7 // indirect
	gopkg.in/yaml.v2 v2.4.0 // indirect
	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
)

replace (
	git.apache.org/thrift.git => github.com/apache/thrift v0.0.0-20180902110319-2566ecd5d999
	github.com/dgrijalva/jwt-go => github.com/golang-jwt/jwt v3.2.1+incompatible
	github.com/microcosm-cc/bluemonday => github.com/microcosm-cc/bluemonday v1.0.15
)
```

</details>

@fasmat 
buffalo fix did not solve the issue :(

```
! This updater will attempt to update your application to Buffalo version: v0.18.0
? Do you wish to continue? [y/n]
y
~~~ Rewriting Imports ~~~
go: finding module for package github.com/gobuffalo/suite/v4
go: downloading github.com/gobuffalo/suite/v4 v4.0.2
go: found github.com/gobuffalo/suite/v4 in github.com/gobuffalo/suite/v4 v4.0.2
go: downloading github.com/gobuffalo/mw-csrf v1.0.0
go: downloading github.com/gobuffalo/validate v2.0.3+incompatible
~~~ Checking webpack.config.js ~~~
~~~ Checking package.json ~~~
~~~ Installing required tools ~~~
~~~ Checking for deprecations ~~~
~~~ Upgrading Dockerfile ~~~
~~~ Removing old plugins ~~~
~~~ Removing github.com/gobuffalo/buffalo-pop plugin ~~~
~~~ Cleaning plugins cache ~~~
~~~ Reinstalling plugins ~~~
/test/test # buffalo build
# test
./buffalo_build_main.go:13:2: imported and not used: "test/models"
./buffalo_build_main.go:14:2: imported and not used: "test/migrations"
Usage:
...
```

Any other hint ?
How can I keep the buffalo_build_main.go file for inspection ?
I will compare with a new project later today/tomorrow

@fasmat 
I am running the tests in a docker container based on go 1.17.3-alpine if this matters

I tried with:
* buffalo 0.18.1 
* cli@development 
* buffalo fix 
 
and the problem is still there

```
DEBU[2021-12-01T02:18:08Z] Exec: go build -v -tags development -o bin/test -ldflags -linkmode external -extldflags "-static"
test/a
test/migrations
test
# test
./buffalo_build_main.go:13:2: imported and not used: "test/models"
./buffalo_build_main.go:14:2: imported and not used: "test/migrations"
DEBU[2021-12-01T02:18:09Z] Rollback: main.go
DEBU[2021-12-01T02:18:09Z] File: /test/test/main.go
DEBU[2021-12-01T02:18:09Z] Delete: a/aa.go
DEBU[2021-12-01T02:18:09Z] Delete: a/embed.go
DEBU[2021-12-01T02:18:09Z] Delete: a/inflections.json
DEBU[2021-12-01T02:18:09Z] Delete: buffalo_build_main.go
DEBU[2021-12-01T02:18:09Z] Delete: migrations/.pop-tmp.md
```

I tried to copy the embed.go file from a new project (I used the file in the locales directory, there was none in the models dir) but I still get the same error (embed.go file below)
```
package models

import (
	"embed"
	"io/fs"

	"github.com/gobuffalo/buffalo"
)

//go:embed *
var files embed.FS

func FS() fs.FS {
	return buffalo.NewFS(files, "models")
}
```

BTW if I try to run a newly generated projecy (```buffalo new xxx```) I get 
```
templates/embed.go:10:14: pattern */*: no matching files found
```

Your error messages which blocks the build are
```
./buffalo_build_main.go:13:2: imported and not used: "test/models"
./buffalo_build_main.go:14:2: imported and not used: "test/migrations"
```
which means the source has import directives for those two application-specific paths but they are not used.

Those two directories are all related to models (~= database) and from your info details, I see this:
```
WithPop     false
```

So your application does not have models and migrations but the main function of the builder imported for them. I just committed a patch and PRed against `development` but is not yet merged.

If you want to test it right now or have time to test, please kindly build the `buffalo` binary from my fork and try to build with it, and will be so much happy if you let us know the result.

https://github.com/gochigo/cli/tree/hotfix-build-fails-without-pop


For the new error message on a newly generated app,
```
templates/embed.go:10:14: pattern */*: no matching files found
```
It is a known issue when happens there is no model generated. If you run `buffalo build` after `buffalo g resource something`, the error will be gone. Or, if you will not have model-specific templates, you need to erase `*/*` part from the `templates/embed.go` file.

@sio4 
I cloned the linked repo and executed
```
git checkout -b hotfix-build-fails-without-pop
go install ./cmd/buffalo
```

The resulting binary is installed in /go/bin

The I re-run buffalo fix and buffalo build and get the same error :(

```
DEBU[2021-12-01T05:07:32Z] Step: a8daa6f6
DEBU[2021-12-01T05:07:32Z] Chdir: /test/test
DEBU[2021-12-01T05:07:33Z] File: /test/test/main.go
DEBU[2021-12-01T05:07:33Z] File: /test/test/a/aa.go
DEBU[2021-12-01T05:07:33Z] File: /test/test/a/embed.go
DEBU[2021-12-01T05:07:33Z] File: /test/test/buffalo_build_main.go
DEBU[2021-12-01T05:07:33Z] File: /test/test/migrations/.pop-tmp.md
DEBU[2021-12-01T05:07:33Z] File: /test/test/migrations/embed.go
DEBU[2021-12-01T05:07:33Z] File: /test/test/a/inflections.json
DEBU[2021-12-01T05:07:33Z] Exec: go mod tidy
DEBU[2021-12-01T05:07:33Z] Exec: go build -v -tags development -o bin/test
test
# test
./buffalo_build_main.go:13:2: imported and not used: "pro/models"
./buffalo_build_main.go:14:2: imported and not used: "pro/migrations"
DEBU[2021-12-01T05:07:34Z] Rollback: main.go
DEBU[2021-12-01T05:07:34Z] File: /test/test/main.go
DEBU[2021-12-01T05:07:34Z] Delete: a/aa.go
DEBU[2021-12-01T05:07:34Z] Delete: a/embed.go
DEBU[2021-12-01T05:07:34Z] Delete: a/inflections.json
DEBU[2021-12-01T05:07:34Z] Delete: buffalo_build_main.go
DEBU[2021-12-01T05:07:34Z] Delete: migrations/.pop-tmp.md
DEBU[2021-12-01T05:07:34Z] Delete: migrations/embed.go
DEBU[2021-12-01T05:07:34Z] Exec: go mod tidy
Usage:
  buffalo build [flags]

Aliases:
  build, b, bill, install
...
```

maybe I am missing something...

--- a/internal/cmd/build/build_test.go
+++ b/internal/cmd/build/build_test.go
@@ -41,6 +41,12 @@ func TestBuild(t *testing.T) {
 			resourceargs: []string{"g", "resource", "phone", "model"},
 			appname:      "sqlite",
 		},
+		{
+			name:         "skipop",
+			newargs:      []string{"new", "skipop", "-f", "--skip-pop", "--vcs", "none"},
+			resourceargs: []string{"g", "action", "phone", "new"},
+			appname:      "skipop",
+		},
 	}
 
 	for _, v := range tcases {
@@ -53,6 +59,10 @@ func TestBuild(t *testing.T) {
 
 				os.Chdir(v.appname)
 
+				// NOTE: I think adding this to here is totally fine since
+				// this is "integration" test. However, the original reason
+				// I added it now is to prevent build failure when there is
+				// no subdir under templates directory (go:embed * */*)
 				out, err = testhelpers.RunBuffaloCMD(t, v.resourceargs)
 				tx.Log(out)
 				r.NoError(err)

--- a/internal/genny/build/templates/buffalo_build_main.go.plush
+++ b/internal/genny/build/templates/buffalo_build_main.go.plush
@@ -12,14 +12,16 @@ import (
 	<%= if (opts.WithGrifts) { %>
 	_ "<%= opts.GriftsPkg %>"
 	<% } %>
+<%= if (opts.WithPop) { %>
 	"<%= opts.ModelsPkg %>"
 	"<%= opts.PackagePkg %>/migrations"
+<% } %>
 
 	"github.com/gobuffalo/buffalo/runtime"
 	"github.com/markbates/grift/grift"
-	<%= if (opts.WithPop) { %>
+<%= if (opts.WithPop) { %>
 	"github.com/gobuffalo/pop/v6"
-	<% } %>
+<% } %>
 )
 
 func init() {
@@ -53,45 +55,45 @@ func main() {
 		return
 	}
 
-  	c := args[1]
-  	switch c {
-  	<%= if (opts.WithPop) { %>
-  	case "migrate":
-    	migrate()
-  	<% } %>
-  	case "version":
-    	printVersion()
-  	case "task", "t", "tasks":
-    	if len(args) < 3 {
-      		log.Fatal("not enough arguments passed to task")
-    	}
-    	
-		c := grift.NewContext(args[2])
-    	if len(args) > 2 {
-      		c.Args = args[3:]
-    	}
-    	
-    	if err := grift.Run(args[2], c); err != nil {
-      		log.Fatal(err)
-    	}
-  	default:
-    	if _, err := exec.LookPath("buffalo"); err != nil {
-        	log.Fatal(err)
-    	}
-    
-		cmd := exec.Command("buffalo", args[1:]...)
-		cmd.Stdin = os.Stdin
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-    	
-    	if err := cmd.Run(); err != nil {
-      		log.Fatal(err)
-    	}
-  	}
+	c := args[1]
+	switch c {
+<%= if (opts.WithPop) { %>
+	case "migrate":
+	migrate()
+<% } %>
+	case "version":
+	printVersion()
+	case "task", "t", "tasks":
+	if len(args) < 3 {
+		log.Fatal("not enough arguments passed to task")
+	}
+
+	c := grift.NewContext(args[2])
+	if len(args) > 2 {
+		c.Args = args[3:]
+	}
+
+	if err := grift.Run(args[2], c); err != nil {
+		log.Fatal(err)
+	}
+	default:
+	if _, err := exec.LookPath("buffalo"); err != nil {
+		log.Fatal(err)
+	}
+
+	cmd := exec.Command("buffalo", args[1:]...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+	}
 }
 
 func printVersion() {
-  	fmt.Printf("<%= opts.Name.Title() %> version %s\n", runtime.Build())
+	fmt.Printf("<%= opts.Name.Title() %> version %s\n", runtime.Build())
 }
 
 <%= if (opts.WithPop) { %>
@@ -100,9 +102,9 @@ func migrate() {
 	if err != nil {
 		log.Fatalf("Failed to unpack migrations: %s", err)
 	}
-  	
-  	if err := box.Up(); err != nil {
-    	log.Fatalf("Failed to run migrations: %s", err)
-  	}
+
+	if err := box.Up(); err != nil {
+	log.Fatalf("Failed to run migrations: %s", err)
+	}
 }
 <% } %>

